scanpy/preprocessing/normalization.py

```diff
@@ -0,0 +1,204 @@
+import numpy as np
+from scipy.sparse import issparse
+from sklearn.utils import sparsefuncs
+from .. import logging as logg
+
+def _normalize_data(X, counts, after=None, copy=False):
+    X = X.copy() if copy else X
+    after = np.median(counts) if after is None else after
+    counts += (counts == 0)
+    counts /= after
+    if issparse(X):
+        X = sparsefuncs.inplace_row_scale(X, 1/counts)
+    else:
+        X /= counts[:, None]
+    return X if copy else None
+
+def normalize_quantile(data, counts_per_cell_after=None, counts_per_cell=None,
+                       quantile=1, min_counts=1, key_n_counts=None, copy=False,
+                       layers=[], use_rep=None):
+    """Normalize total counts per cell.
+
+    Normalize each cell by total counts over genes, so that every cell has
+    the same total count after normalization.
+
+    Similar functions are used, for example, by Seurat [Satija15]_, Cell Ranger
+    [Zheng17]_ or SPRING [Weinreb17]_.
+
+    Parameters
```

Comment:
Can we please stick to 80 lines also in the docstrings? On my 13inch this already gets messed up here when looking at the file on Github, so that I cannot properly check this.